### PR TITLE
[SG-37461]: logging(gitstart): migrate internal/repos to sourcegraph/log

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Task URL
https://app.gitstart.com/tasks/64396

## Test plan

##Credit cost
70 credits

### CODE REVIEW

File: cmd/frontend/graphqlbackend/affiliated_repositories_connection.go [hunk 1]
```diff
@@ -82,7 +84,7 @@ func (a *affiliatedRepositoriesConnection) getNodesAndErrors(ctx context.Context
 		)
 		for _, svc := range svcs {
 			svcsByID[svc.ID] = svc
-			src, err := repos.NewSource(ctx, a.db, svc, cf)
+			src, err := repos.NewSource(ctx, log.Scoped("getNodesAndErrors", ""), a.db, svc, cf)
```
Comment (bobheadxi) [inline cmd/frontend/graphqlbackend/affiliated_repositories_connection.go:87]:
```suggestion
			src, err := repos.NewSource(ctx, log.Scoped("repos.source", ""), a.db, svc, cf)
```

File: cmd/frontend/graphqlbackend/external_service.go [hunk 2]
```diff
@@ -187,7 +187,7 @@ func (r *externalServiceResolver) NextSyncAt() *DateTime {
 var scopeCache = rcache.New("extsvc_token_scope")
 
 func (r *externalServiceResolver) GrantedScopes(ctx context.Context) (*[]string, error) {
-	scopes, err := repos.GrantedScopes(ctx, scopeCache, r.db, r.externalService)
+	scopes, err := repos.GrantedScopes(ctx, log.Scoped("GrantedScopes", ""), scopeCache, r.db, r.externalService)
```
Comment (bobheadxi) [inline cmd/frontend/graphqlbackend/external_service.go:190]:
```suggestion
	scopes, err := repos.GrantedScopes(ctx, r.logger.Scoped("GrantedScopes", ""), scopeCache, r.db, r.externalService)
```

Comment (0xedb) [reply cmd/frontend/graphqlbackend/external_service.go]: fixed

File: cmd/frontend/graphqlbackend/affiliated_repositories_connection.go (hunk 1 above)
Comment (0xedb) [reply cmd/frontend/graphqlbackend/affiliated_repositories_connection.go]: fixed

File: internal/repos/main_test.go [hunk 3]
```diff
@@ -25,7 +23,6 @@ func update(name string) bool {
 func TestMain(m *testing.M) {
 	flag.Parse()
 	if !testing.Verbose() {
-		log15.Root().SetHandler(log15.DiscardHandler())
```
Comment (gitstart-sourcegraph) [inline internal/repos/main_test.go:28]:
```go
logtest.InitWithLevel(m, log.LevelNone)
```

File: cmd/repo-updater/shared/main.go [hunk 4]
```diff
@@ -195,14 +195,14 @@ func Main(enterpriseInit EnterpriseInit) {
...
-	go repos.RunPhabricatorRepositorySyncWorker(ctx, store)
+	go repos.RunPhabricatorRepositorySyncWorker(ctx, log.Scoped("shared.Main", ""), store)
 
 	// git-server repos purging thread
 	var purgeTTL time.Duration
 	if envvar.SourcegraphDotComMode() {
 		purgeTTL = 14 * 24 * time.Hour // two weeks
 	}
-	go repos.RunRepositoryPurgeWorker(ctx, db, purgeTTL)
+	go repos.RunRepositoryPurgeWorker(ctx, log.Scoped("shared.Main", ""), db, purgeTTL)
```
Comment (bobheadxi) [inline cmd/repo-updater/shared/main.go:205]: Why is this scope `shared.Main`? shouldn't it be `RepositoryPurgeWorker`? Entries logged by this logger belong to that, not `Main`

File: enterprise/internal/batches/sources/sources.go [hunk 5]
```diff
@@ -439,7 +441,7 @@ func extractCloneURL(ctx context.Context, s database.ExternalServiceStore, repo
 	for _, svc := range svcs {
 		// build the clone url using the external service config instead of using
 		// the source CloneURL field
-		cloneURL, err := repos.CloneURL(svc.Kind, svc.Config, repo)
+		cloneURL, err := repos.CloneURL(log.Scoped("sources.extractCloneURL", ""), svc.Kind, svc.Config, repo)
```
Comment (bobheadxi) [inline enterprise/internal/batches/sources/sources.go:444]: same here

```suggestion
		cloneURL, err := repos.CloneURL(log.Scoped("CloneURL", ""), svc.Kind, svc.Config, repo)
```

File: cmd/repo-updater/repoupdater/server.go [hunk 6]
```diff
@@ -203,7 +203,7 @@ func (s *Server) handleExternalServiceSync(w http.ResponseWriter, r *http.Reques
 	if sourcer = s.Sourcer; sourcer == nil {
 		db := database.NewDBWith(logger, s)
 		depsSvc := livedependencies.GetService(db, nil)
-		sourcer = repos.NewSourcer(db, httpcli.ExternalClientFactory, repos.WithDependenciesService(depsSvc))
+		sourcer = repos.NewSourcer(s.Logger.Scoped("handleExternalServiceSync", ""), db, httpcli.ExternalClientFactory, repos.WithDependenciesService(depsSvc))
```
Comment (bobheadxi) [inline cmd/repo-updater/repoupdater/server.go:206]: same as the other suggestions - please audit scopes, scopes should apply to what is going to get logged

```suggestion
		sourcer = repos.NewSourcer(s.Logger.Scoped("repos.Sourcer", ""), db, httpcli.ExternalClientFactory, repos.WithDependenciesService(depsSvc))
```

File: internal/repos/sources.go [hunk 7]
```diff
@@ -39,24 +41,24 @@ func NewSourcer(db database.DB, cf *httpcli.Factory, decs ...func(Source) Source
...
-		return NewGithubSource(externalServicesStore, svc, cf)
+		return NewGithubSource(logger, externalServicesStore, svc, cf)
 	case extsvc.KindGitLab:
-		return NewGitLabSource(ctx, db, svc, cf)
+		return NewGitLabSource(ctx, logger, db, svc, cf)
 	case extsvc.KindGerrit:
 		return NewGerritSource(svc, cf)
 	case extsvc.KindBitbucketServer:
-		return NewBitbucketServerSource(svc, cf)
+		return NewBitbucketServerSource(logger, svc, cf)
```
Comment (jhchabran) [inline internal/repos/sources.go:55]: This constructor is not scoping the logger on its own, so it should be scoped here.

File: internal/repos/sources.go [hunk 8]
```diff
@@ -39,24 +41,24 @@ func NewSourcer(db database.DB, cf *httpcli.Factory, decs ...func(Source) Source
...
 	case extsvc.KindBitbucketCloud:
-		return NewBitbucketCloudSource(svc, cf)
+		return NewBitbucketCloudSource(logger, svc, cf)
 	case extsvc.KindGitolite:
 		return NewGitoliteSource(db, svc, cf)
 	case extsvc.KindPhabricator:
-		return NewPhabricatorSource(svc, cf)
+		return NewPhabricatorSource(logger, svc, cf)
```
Comment (jhchabran) [inline internal/repos/sources.go:61]: This constructor is not scoping the logger on its own, so it should be scoped here.

File: internal/repos/sources.go [hunk 9]
```diff
@@ -39,24 +41,24 @@ func NewSourcer(db database.DB, cf *httpcli.Factory, decs ...func(Source) Source
...
+func NewSource(ctx context.Context, logger log.Logger, db database.DB, svc *types.ExternalService, cf *httpcli.Factory) (Source, error) {
 	externalServicesStore := db.ExternalServices()
 
 	switch strings.ToUpper(svc.Kind) {
 	case extsvc.KindGitHub:
-		return NewGithubSource(externalServicesStore, svc, cf)
+		return NewGithubSource(logger, externalServicesStore, svc, cf)
 	case extsvc.KindGitLab:
-		return NewGitLabSource(ctx, db, svc, cf)
+		return NewGitLabSource(ctx, logger, db, svc, cf)
```
Comment (jhchabran) [inline internal/repos/sources.go:51]: This constructor is not scoping the logger on its own, so it should be scoped here.

File: internal/repos/syncer.go [hunk 10]
```diff
@@ -80,7 +80,7 @@ func (s *Syncer) Run(ctx context.Context, store Store, opts RunOptions) error {
 		s.initialUnmodifiedDiffFromStore(ctx, store)
 	}
 
-	worker, resetter := NewSyncWorker(ctx, store.Handle(), &syncHandler{
+	worker, resetter := NewSyncWorker(ctx, s.Logger, store.Handle(), &syncHandler{
```
Comment (jhchabran) [inline internal/repos/syncer.go:83]:
```suggestion
	worker, resetter := NewSyncWorker(ctx, s.Logger.Scoped("syncWorker", ""), store.Handle(), &syncHandler{
```

**Review by jhchabran — CHANGES_REQUESTED** [review]
A few remarks, but we're getting there!

**Review by jhchabran — APPROVED** [review]

**Pull request merged**